packages/loopring_v3/contracts/iface/IExchangeV3.sol

```diff
@@ -889,84 +778,110 @@ contract IExchangeV3 is IExchange
+    /// @param  owner The address of the account the withdrawal was done for.
+    /// @param  token The token address
     function withdrawFromApprovedWithdrawal(
-        uint blockIdx,
-        uint slotIdx
+        address owner,
+        address token
         )
         external;
 
-    /// @dev Allows the operator to withdraw the fees he earned by processing the
-    ///      deposit and onchain withdrawal requests.
+    /// @dev Gets the amount that can be withdrawn immediately with `withdrawFromApprovedWithdrawal`.
+    /// @param  owner The address of the account the withdrawal was done for.
+    /// @param  token The token address
+    /// @return The amount withdrawable
+    function getAmountWithdrawable(
+        address owner,
+        address token
+        )
+        external
+        view
+        returns (uint);
+
+    // -- Agents --
+
+    /// @dev Authorizes/Deauthorizes agents for an account.
+    ///      An agent is allowed to authorize onchain operations for the account owner.
+    ///      By definition the account owner is an agent for himself.
     ///
-    ///      This function is only callable by the exchange operator.
+    ///      This function can only be called by an agent.
     ///
-    ///      The block fee can only be withdrawn from finalized blocks
-    ///      (i.e. blocks that can never be reverted).
+    /// @param owner The account owner.
+    /// @param agents The agents to be authorized/deauthorized.
+    /// @param authorized True to authorize the agent, false to deauthorize
+    function authorizeAgents(
+        address   owner,
+        address[] calldata agents,
+        bool[]    calldata authorized
+        )
+        external;
+
+    /// @dev Returns whether an agent address is an agent of an account owner
+    /// @param owner The account owner.
+    /// @param agent The agent address
+    /// @return True if the agent address is an agent for the account owner, else false
+    function isAgent(address owner, address agent)
+        public
+        view
+        returns (bool);
+
+    /// @dev Approves an internal transfer.
+    ///      Important! This is just an approval, the operator has full control
+    ///      whether the transfer will actally be done!
     ///
-    /// @param  blockIdx The block index to withdraw the funds for
-    /// @param  feeRecipient The address that receives the block fee
-    /// @return feeAmount The amount of ETH earned in the block and sent to the operator
-    function withdrawBlockFee(
-        uint    blockIdx,
-        address payable feeRecipient
+    ///      This function can only be called by an agent.
+    ///
+    /// @param from The address of the account that sends the tokens.
+    /// @param to The address to which 'amount' tokens are transferred.
+    /// @param token The address of the token to transfer ('0x0' for ETH).
+    /// @param amount The amount of tokens to be transferred.
+    function approveConditionalTransfer(
```

Comment:
it doesn't seem there is a *condition* here, so maybe simply **approveOffchainTransfer**?